src/utils/Pausable.huff

```diff
@@ -0,0 +1,126 @@
+/// @title Pausable
+/// @notice SPDX-License-Identifier: MIT
+/// @author zarf.eth <https://github.com/zarfsec>
+/// @notice A Pausable implementation
+
+#include "../auth/NonPayable.huff"
+
+
+// Interface
+
+/// @notice returns whether the contract is paused
+#define function isPaused() view returns (bool)
+
+/// @notice Pauses the contract 
+/// @dev Only callable when the contract is unpaused.
+#define function pause() nonpayable returns ()
+
+/// @notice Unpauses the contract 
+/// @dev Only callable when the contract is paused.
+#define function unpause() nonpayable returns ()
+
+/// @notice Emitted when contract is paused.
+#define event Paused(address)
+
+/// @notice Emitted when contract is unpaused.
+#define event Unpaused(address)
+
+
+
+// Storage
+
+/// @notice Paused Storage Slot
+#define constant PAUSED_SLOT = FREE_STORAGE_POINTER()
+
+/// @notice Unpaused representation
+#define constant _UNPAUSED = 0x01
+
+/// @notice Paused representation
+#define constant _PAUSED = 0x02
+
+/// @notice cast keccak "Paused(address)"
+#define constant EVENT_PAUSED = 0x62e78cea01bee320cd4e420270b5ea74000d11b0c9f74754ebdbfc544b05a258
```

Comment:
we can replace these with the [__EVENT_HASH](https://docs.huff.sh/get-started/huff-by-example/#event-hash-event-def-string) builtin

Reply:
Didn't know about this builtin function :) Implemented!